sycl/source/detail/kernel_impl.cpp

```diff
@@ -93,6 +93,35 @@ bool kernel_impl::isCreatedFromSource() const {
   return MCreatedFromSource;
 }
 
+void kernel_impl::checkIfValidForNumArgsInfoQuery() const {
+  bool isBuiltIn = false;
+  std::string KernelName = get_info<info::kernel::function_name>();
+  if (KernelName != "") {
+    auto Devices = MKernelBundleImpl->get_devices();
+    for (const auto &Device : Devices) {
+      auto BuiltInKernles =
+          Device.get_info<info::device::built_in_kernel_ids>();
+      if (!BuiltInKernles.empty()) {
+        auto It = std::find_if(BuiltInKernles.begin(), BuiltInKernles.end(),
+                               [&KernelName](kernel_id &Id) {
+                                 return Id.get_name() == KernelName;
+                               });
+        if (It != BuiltInKernles.end()) {
+          isBuiltIn = true;
+          break;
+        }
+      }
+    }
+  }
+  if (!MKernelBundleImpl->isInterop() && !isBuiltIn)
```

Comment:
Let's insert an early exit from this function if kernel bundle is an interop kernel bundle. I understand that it won't be taken at this moment, but it will make more impact later.

Reply:
Makes sense, and hence we don't need the `isBuiltIn` variable.